src/webportal/src/app/job-submission/components/data/data-component.jsx

```diff
@@ -32,34 +43,116 @@ function reducer(state, action) {
 
 export const DataComponent = React.memo((props) => {
   const hdfsClient = new WebHDFSClient(host);
+  const api = config.restServerUri;
+  const user = cookies.get('user');
+  const token = cookies.get('token');
 
   const {onChange} = props;
-  const [jobData, dispatch] = useReducer(reducer, new JobData(hdfsClient, [], null));
+  const [teamConfigs, setTeamConfigs] = useState();
+  const [defaultTeamConfigs, setDefaultTeamConfigs] = useState();
+  const [jobData, dispatch] = useReducer(
+    reducer,
+    new JobData(hdfsClient, [], null),
+  );
+
+  useEffect(() => {
+    const teamStorageClient = new TeamStorageClient(api, user, token);
+    const userGroupPromise = teamStorageClient.fetchUserGroup();
+    const configPromise = teamStorageClient.fetchStorageConfigData();
+    const serverPromise = teamStorageClient.fetchStorageServer();
+    Promise.all([userGroupPromise, configPromise, serverPromise]).then(
+      ([userGroups, storageConfigData, storageServerData]) => {
+        const newConfigs = [];
```

Comment:
Move this piece of code to another function and give it a meaningful name? Not easy to understand

Reply:
It is really hard to understand due to its complex logic. It's hard to refactor because it relies on the component states, and I am afraid put these codes into a function would not make it more clear. @sunqinzheng any suggestions?